src/Orleans.Transactions/InClusterTM/TransactionManager.cs

```diff
@@ -5,10 +5,99 @@
 using System.Threading;
 using Microsoft.Extensions.Options;
 using Microsoft.Extensions.Logging;
+using Orleans.Runtime;
 using Orleans.Transactions.Abstractions;
+using Orleans.Runtime.Configuration;
 
 namespace Orleans.Transactions
 {
+    internal class TransactionManagerMetrics
```

Comment:
I sense some duplicated code with `TransactionAgentMetrics`. Any reason to not have a common implementation?

Reply:
The structure is similar, but metrics are different. They can share a common base class. I can improve that.

Reply:
I look into this direction, and have some more thoughts . These two metrics are similar in structure because they all use PeriodicAction to trigger metric recording, which is already made modular to increase code reuse. Logic other than PeriodicAction class are only similar in terms of function name, the actual logic in the function is different for each metric class. So not much upside on making a common implementation. 

And to make a common implementation,  we need to have a public class in Orleans runtime. I'm wary on this one, since I don't want to signal our users this is the recommended way to produce metrics. There's many many ways, and it is up to them on which way they choose to use.